src/app/util/services/lu-core-data.service.ts

```diff
@@ -79,22 +79,86 @@ export class LuCoreDataService {
     this.$apiUrl.next(new ApiConfig({ base, user, pass }));
   }
 
-  get(url: string): Observable<any> {
+  get(url: string, responseType?: "text", cachedTransformation?: OperatorFunction<any, any>): Observable<any> {
     if (!this.cache.has(url)) {
       let httpRequest = this.$apiUrl.pipe(switchMap(cfg => this.http.get(cfg.base + url, {
         headers: cfg.headers(),
+        responseType: responseType as "json",
```

Comment:
This is weird, as if you try to force the typecheck when it doesn't fit

Reply:
Yeah, I encountered weird issues in testing when I set the parameter type to `'text' | 'json'", so I added that. I've now checked and it appears this was due to angular defining multiple different method signatures with the same name but with differing return types based on the value of `responseType`. Now that the parameter is just `'text'` it turns out the typecheck has no issues anymore, so I removed the typecast now.

Reply:
Does that not break the other JSON endpoints?

Reply:
Apparently not, it's either a null or "text" which the compiler apparently can tell apart properly, and null makes it use json parsing, so it's backwards compatible.